### DISCUSSION TITLE
Issues Importing Custom Typings from Private Layer in Nuxt Project

### DISCUSSION THREAD

**Discussion by hamedfaryabi** [nuxt/nuxt#29796] [Questions]
**Question: Issues Importing Custom Typings from Private Layer in Nuxt Project**

Hello,

I'm working with a custom Nuxt layer named "PMS" that I published on a private Git repository. This layer includes type declarations in a file located at `types/pms.d.ts`, structured as follows:

```typescript
declare module "pms" {
  type FormErrors<T = string> = Record<T, string[]>;
  interface Tab {
    name: string;
    title: string;
    show?: boolean;
  }
}
```

When I try to use these types in my main project with the following import:

```typescript
import type { FormErrors } from 'pms';
```

I receive an error in both VSCode and WebStorm:  
`Cannot find module 'pms' or its corresponding type declarations.`

### Troubleshooting Steps

1. **Nuxt Config**: I added the following in `nuxt.config.ts` to auto-import types:

   ```typescript
   imports: {
     autoImport: true,
     dirs: [
       "types/**",
     ],
   },
   ```

2. **Layer Package.json**: I updated `package.json` with these fields:

   ```json
   {
     "main": "./nuxt.config.ts",
     "types": "./types/pms.d.ts",
     "files": [
       "./nuxt.config.ts",
       "types/pms.d.ts"
     ]
   }
   ```

Despite these changes, the error persists. I would appreciate any advice on correctly configuring the layer or setting up the type imports so that the `FormErrors` and other types are recognized.

Thank you for your help!

**Comment by luigigorlero** [comment]
I came across this when trying to figure out the same thing, I ended up using the `alias` config option. My setup is:

1. A `./types/index.ts` file where I export all typings
2. In my **layer’s** `nuxt.config.ts` I’m configuring an alias for my layer
```ts
import { dirname, join } from 'node:path'
import { fileURLToPath } from 'node:url'

const currentDir = dirname(fileURLToPath(import.meta.url))

export default defineNuxtConfig({
  alias: {
    'my-layer': join(currentDir, './types'),
  },
})
```
3. This allows any component within your layer and your main app to access anything exported in that `./types/index.ts` using
`import { Class, type Type } from 'my-layer'`
